### ISSUE TITLE
Dataset Creator

### ISSUE THREAD

**Issue by amitbcp** [mlmed/torchxrayvision#22]
Thanks for the wonderful repo. I had some ideas to add flexibility to datasets loader for further experiments .

1. For CheXpert datasets, the pure_label option is not available. My current understanding is, pure labels for other datasets filter those scans with only 1 finding. Currently for CheXpert the loader only returns Multi-labelled examples.
2. Question/Doubt : How are the NaNs getting handled for CheXpert ? Does the Dataloader converts converts NaNs to 0 ?
3. The pathologies for all the Datasets are hard coded. Maybe we can provide user with an option to chose pathologies from each datasets. Like maybe including a "No Finding" if they want 

Do share your thoughts on the same. Let me know if you would me to raise a PR for some of it ?

**Comment by ieee8023** [comment]
> pure_labels

Yes needs to be implemented for all loaders. I'll put it on my to-do list!

> NaNs

NaNs are returned from the dataloader if the label was -1. Then in the training they are not counted in the losses. We cannot assume a value for these.

> Pathologies
Right now the idea is that a dataset will populate itself with all available labels and then xrv.datasets.relabel_datasets will reduce the labels to those specified by a list (or add NaNs if it doesn't exist) as seen here: https://github.com/mlmed/torchxrayvision/blob/master/scripts/xray_datasets.ipynb
I think No Finding is a tricky label so I didn't add it. Instead I used it to indicate all the other labels were 0. It could be added to the default list of pathologies for the datasets that have it.

**Comment by ieee8023** [comment]
I'm closing this for now. Let me know if something should still be worked on. 

I think for the pure_labels it is better to offload it to the user to construct a dataset and then use a SubsetDataset to filter the dataset based on processing the .labels to find the samples with a specific criteria. PR: https://github.com/mlmed/torchxrayvision/pull/86
